src/probnum/filtsmooth/gaussian/approx/_unscentedkalman.py

```diff
@@ -1,263 +1,156 @@
+        return _linearize_via_cubature(
             t=t,
-            dt=dt,
-            compute_gain=compute_gain,
-            _diffusion=_diffusion,
-            _linearise_at=_linearise_at,
+            model=self.non_linear_model,
+            rv=at_this_rv,
+            unit_params=self._cubature_params,
+            forw_impl=self._forward_implementation,
+            backw_impl=self._backward_implementation,
         )
 
-    def forward_rv(
-        self, rv, t, dt=None, compute_gain=False, _diffusion=1.0, _linearise_at=None
-    ) -> Tuple[randvars.Normal, Dict]:
-        raise NotImplementedError("TODO")  # Issue  #234
 
-    def backward_realization(
-        self,
-        realization_obtained,
-        rv,
-        rv_forwarded=None,
-        gain=None,
-        t=None,
-        dt=None,
-        _diffusion=1.0,
-        _linearise_at=None,
-    ):
-        return self._backward_realization_as_rv(
-            realization_obtained,
-            rv=rv,
-            rv_forwarded=rv_forwarded,
-            gain=gain,
-            t=t,
-            dt=dt,
-            _diffusion=_diffusion,
-            _linearise_at=_linearise_at,
-        )
+def _spherical_cubature_unit_params(*, dim):
+    """Return sigma points and weights for spherical cubature integration.
 
-    def backward_rv(
-        self,
-        rv_obtained,
-        rv,
-        rv_forwarded=None,
-        gain=None,
-        t=None,
-        dt=None,
-        _diffusion=1.0,
-        _linearise_at=None,
-    ):
-        raise NotImplementedError("Not available (yet).")
-
-
-class DiscreteUKFComponent(UKFComponent, randprocs.markov.discrete.NonlinearGaussian):
-    """Discrete unscented Kalman filter transition."""
+    Reference:
+    Bayesian Filtering and Smoothing. Simo Särkkä. Page 111.
+    """
+    s, I, zeros = np.sqrt(dim), np.eye(dim), np.zeros((1, dim))
+    unit_sigma_points = s * np.concatenate((zeros, I, -I), axis=0)
+    weights_mean, weights_cov = _weights(dim)
+    return unit_sigma_points, (weights_mean, weights_cov)
 
-    def __init__(
-        self,
-        non_linear_model,
-        spread: Optional[FloatLike] = 1e-4,
-        priorpar: Optional[FloatLike] = 2.0,
-        special_scale: Optional[FloatLike] = 0.0,
-    ) -> None:
-        UKFComponent.__init__(
-            self,
-            non_linear_model,
-            spread=spread,
-            priorpar=priorpar,
-            special_scale=special_scale,
-        )
 
-        randprocs.markov.discrete.NonlinearGaussian.__init__(
-            self,
-            input_dim=non_linear_model.input_dim,
-            output_dim=non_linear_model.output_dim,
-            transition_fun=non_linear_model.transition_fun,
-            transition_fun_jacobian=non_linear_model.transition_fun_jacobian,
-            noise_fun=non_linear_model.noise_fun,
-        )
+def _weights(dim):
+    spread, priorpar, special_scale = 1.0, 0.0, 0.0
+    scale = spread**2 * (dim + special_scale) - dim
```

Comment:
as discussed offline, these parameters are not really part of the API anymore. But if we decide later on that they are important, here is where they _could_ be added. Let's do all the refactoring first, anyway.